src/cdk/layout/breakpoints-observer.ts

```diff
@@ -76,7 +76,7 @@ export class BreakpointObserver implements OnDestroy {
     const observables = queries.map(query => this._registerQuery(query).observable);
 
     return combineLatest(observables).pipe(
-      debounceTime(0, asapScheduler),
+      debounceTime(0),
```

Comment:
In further testing, this delayed values too late in the component lifecycle. Components may depend on the values provided from startWith on line 116. When delayed with the async scheduler, these values were not yet available. Therefore, it would be necessary to skip the first and debounce only the subsequent emissions.

const stateObservable = combineLatest(observables);
stateObservable = concat(
  stateObservable.pipe(first()),
  stateObservable.pipe(skip(1), debounceTime(0)));
return stateObservable.pipe(map((breakpointStates: InternalBreakpointState[]) => { ... }));

Unfortunately, this emits once and then breaks. I am working to determine the problem.

Reply:
There is another bug in 7.3.x due to the variable scope of queryListener on line 104 that breaks the observer after unsubscribing. This is fixed in 8.*, so I did not post a PR. Emitting immediately and debouncing subsequent emissions does seem to fix the issue. It also eliminates the necessity to flush macrotasks in the tests.